triples-generator/src/test/scala/io/renku/triplesgenerator/events/consumers/tsmigrationrequest/migrations/reprovisioning/ReProvisionJudgeSpec.scala

```diff
@@ -42,59 +41,32 @@ class ReProvisionJudgeSpec extends AnyWordSpec with should.Matchers with MockFac
     "return true when TS schema version cannot be found" in new TestCase {
       (versionPairFinder.find _).expects().returning(Option.empty[RenkuVersionPair].pure[Try])
 
-      judge(NonEmptyList.one(renkuVersionPairs.generateOne)).reProvisioningNeeded() shouldBe true.pure[Try]
+      judge(compatibilityGen.generateOne).reProvisioningNeeded() shouldBe true.pure[Try]
     }
 
     "return true when TS schema version is different than the matrix schema version" in new TestCase {
       val tsVersionPair = renkuVersionPairs.generateOne
       (versionPairFinder.find _).expects().returning(tsVersionPair.some.pure[Try])
 
-      val matrixVersionPairs = renkuVersionPairs.generateNonEmptyList(min = 2)
+      val configVersionPairs = compatibilityGen
+        .suchThat(_.schemaVersion != tsVersionPair.schemaVersion)
+        .map(_.copy(reProvisioningNeeded = true))
 
-      judge(matrixVersionPairs).reProvisioningNeeded() shouldBe true.pure[Try]
+      judge(configVersionPairs.generateOne).reProvisioningNeeded() shouldBe true.pure[Try]
     }
 
-    "return true when the top two schema versions in the matrix are the same " +
```

Comment:
Maybe it would be also good to add a test (I couldn't find one) where will be expecting `false` when versions match but the `reProvisioningNeeded` is `true`.

Reply:
I think we also miss a test that checks the judge to return `false` for different `cliVersion`s and the `reProvisioningNeeded=false`

Reply:
I think we have the last one at least covered in some of the tests that also mention `underReprovisioning` - but it's hard to see and I added these two cases explicitely.